src/adapter/src/coord.rs

```diff
@@ -3316,6 +3316,22 @@ impl<S: Append + 'static> Coordinator<S> {
 
         self.validate_timeline(depends_on.clone())?;
 
+        // Recorded views are not allowed to depend on log sources, as replicas
+        // are not producing the same definite collection for these.
+        // TODO(teskje): Remove this check once arrangement-based log sources
+        // are replaced with persist-based ones.
+        let log_names = depends_on
+            .iter()
+            .flat_map(|id| self.catalog.log_dependencies(*id))
+            .map(|id| self.catalog.get_entry(&id).name().item.clone())
```

Comment:
I think it would be better if this `clone` was be called inside the conditional. That way we only do allocations if there is an error.

Reply:
Isn't this still the case with this implementation? If there is no error, it means that `log_names` is empty, which means that the `flat_map` returns nothing and the `map` body is never called.

I'm not opposed to changing this if you think it would be more readable. But it does add some boilerplate for converting the `Vec<&String>` into a `Vec<String>` afterwards.

Reply:
Oh yep you're 100% right. Seems fine to me then.